### DISCUSSION TITLE
Can you publish this plugin to ChatGPT?

### DISCUSSION THREAD

**Discussion by tenedor** [lean-dojo/LeanDojoChatGPT#3] [General]
I'd love to try out your plug-in but I can't.

ChatGPT Plus users get immediate access to all published plug-ins. Plug-in developer access is blocked on a waitlist and people have been waiting months without getting in.

Publishing a plug-in is apparently very fast once there is a persisted web API. E.g. @namin created and published a Z3 plugin with hours of work, and ChatGPT approved it a week later. If you'd like help, I'm happy to collaborate, I just don't have plug-in developer access.

Thanks!

**Comment by yangky11** [comment]
Hi @tenedor,

Thanks for the suggestion! Just to clarify, do people need plug-in developer access in order to try the code in this repo locally? I have developer access, but I thought ChatGPT pro was sufficient.

Also, I'm wondering what resources do we need to publish this plugin. At least we would need to host LeanDojo somewhere, e.g., AWS?

**Comment by tenedor** [comment]
@yangky11 I could definitely be missing something, but I don't see a way to try the plug-in using ChatGPT plus when I don't have developer access. Here's what I see:

https://github.com/lean-dojo/LeanDojoChatGPT/assets/1499987/4a6caace-a0c2-4f72-9a4c-ef8885b55ccf

Yes, publishing requires basic hosting. Do you have the budget or existing hosting infrastructure to support this? Let me know if you need help on this.

In the future if the plugin gets popular, there may eventually be scaling issues to consider, but I'd wait to cross that bridge when you get to it. Adding simple analytics to count how many hits the API gets is a nice-to-have.

**Comment by yangky11** [comment]
I see, thanks! Unfortunately, we do not have existing infrastructures for hosting web services. We also probably won't have the bandwidth to work on publishing the plugin in this repo, but we're OK with anyone else (with developer access) publishing it. It's a simple wrapper of LeanDojo anyway.

**Comment by tenedor** [comment]
Okay good to know. I can't sign up to do it in the near-term given my current commitments but if I get some time back elsewhere I'll see what I can do.

**Comment by namin** [comment]
Hi!

I created a hosted version, although it only has one worker because of the stateful action that happens in main. So be gentle with it! I also think it will mess up if there are more than one users trying it out at the same time. With that said...

Here is the URL to import LeanDojo ChatGPT as a new GPT action:
https://leandojochatgpt.livecode.ch/openapi.yaml
There is no need to deploy as a plugin, although
https://leandojochatgpt.livecode.ch/
should deploy as one but it doesn't seem to be working right now (and localhost doesn't work either).
There is no need for special access for the new GPTs.

Here is an example run:
https://chat.openai.com/share/b5d4a69e-be18-4caa-995e-2d7eaf0e27ac

Here is the code:
https://github.com/namin/LeanDojoChatGPT/tree/deploy

@tenedor if you want to help out, the next step is to make the repo and commit logic part of the URL so that ChatGPT can provide any one based on the user input. This is how it's done for https://io.livecode.ch/:
https://github.com/namin/io-chatgpt.livecode.ch/blob/main/main.py#L11
The openapi.yaml will also need to change.

Thanks!

**Comment by tenedor** [comment]
@namin nice example. The livecode.ch URIs are giving 500 errors for me. I'm about to go on leave through 11/29 (medical support for a friend), but when I return I'd be happy to dive in on remaining work here. I'll check back in then.

**Comment by namin** [comment]
I'll investigate the errors, thanks.

I have some questions about using a custom repo with LeanDojo -- we can look into that when you're back.

**Comment by tenedor** [comment]
Okay @namin, I'm back! I can replicate your hello world query in my own GPT. Yes, being able to query custom repos would be useful. Would you be up for pairing on this to bring me up to speed? Else I can take a stab on my own.

**Comment by namin** [comment]
@tenedor, cool, welcome back. I am happy to pair on this. I'll email you to set up a time! Thanks :)
